src/Microsoft.VisualStudio.Threading/AsyncReaderWriterLock.cs

```diff
@@ -365,7 +365,7 @@ protected virtual bool CanCurrentThreadHoldActiveLock
         /// Gets a value indicating whether the current SynchronizationContext is one that is not supported
         /// by this lock.
         /// </summary>
-        private static bool IsUnsupportedSynchronizationContext
+        protected virtual bool IsUnsupportedSynchronizationContext
```

Comment:
I don't see any test for this change. Can you add one?

Reply:
Yes, I update the test to make sure the task scheduler can use its special SynchronizationContext.